libraries/Filter/HarmonicNotchFilter.cpp

```diff
@@ -262,6 +296,7 @@ T HarmonicNotchFilter<T>::apply(const T &sample)
     }
 
     T output = sample;
+    WITH_SEMAPHORE(_sem);
```

Comment:
I actually don't think you need a semaphore here. although update() and apply() are called from different threads, there is already an IMU semaphore preventing these happening at the same time. I think it would be better to leave this up to the caller to make thread safe

Reply:
Protection is here - https://github.com/ArduPilot/ardupilot/blob/ad6bf648e90743867c3db4b2dc2c7109b24a8601/libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp#L707 and here https://github.com/ArduPilot/ardupilot/blob/ad6bf648e90743867c3db4b2dc2c7109b24a8601/libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp#L282

Reply:
you're right, removed and comment added